src/lib/storage/table.cpp

```diff
@@ -95,6 +94,36 @@ Chunk& Table::get_chunk(ChunkID chunk_id) {
 
 const Chunk& Table::get_chunk(ChunkID chunk_id) const { return get_chunk(chunk_id); }
 
-void Table::compress_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::compress_chunk"); }
+void Table::compress_chunk(ChunkID chunk_id) {
+  const auto& old_chunk = get_chunk(chunk_id);
+
+  // new empty chunk
+  auto new_chunk = std::make_unique<Chunk>();
+
+  std::vector<std::thread> threads;
+  threads.reserve(old_chunk.size());
+
+  for (ColumnID column_id(0); column_id < old_chunk.size(); ++column_id) {
+    const auto base_segment = old_chunk.get_segment(column_id);
+    const auto segment_type = _column_types[column_id];
+
+    std::shared_ptr<BaseSegment> compressed_segment;
+
+    threads.push_back(std::thread([&compressed_segment, &base_segment, &segment_type]() {
```

Comment:
Won't the captured references go out of scope?

Reply:
It won't since they join the threads in 122, before these local vars are destructed. But I agree that this threading logic is hard to verify.

Reply:
The variables are local to the loop, so they will be destructed beforehand.